server/setup_flow.go

```diff
@@ -0,0 +1,453 @@
+	keyDelegatedFromUserID = "DelegatedFromUserID"
+	keyDelegatedTo         = "Delegated"
+	keyEdition             = "Edition"
+	keyJiraURL             = "JiraURL"
+	keyManageWebhooksURL   = "ManageWebhooksURL"
+	keyMattermostKey       = "MattermostKey"
+	keyPluginURL           = "PluginURL"
+	keyPublicKey           = "PublicKey"
+	keyWebhookURL          = "WebhookURL"
+)
+
+func (p *Plugin) NewSetupFlow() *flow.Flow {
+	pluginURL := *p.client.Configuration.GetConfig().ServiceSettings.SiteURL + "/" + "plugins" + "/" + manifest.ID
+	conf := p.getConfig()
+	return flow.NewFlow("setup-wizard", p.client, pluginURL, conf.botUserID).
+		WithSteps(
+			p.stepWelcome(),
+			p.stepDelegate(),
+			p.stepDelegated(),
+			p.stepDelegateComplete(),
+			p.stepChooseEdition(),
+
+			// Jira Cloud steps
+			p.stepCloudAddedInstance(),
+			p.stepCloudEnableDeveloperMode(),
+			p.stepCloudUploadApp(),
+			p.stepCloudInstalledApp(),
+
+			// Jira server steps
+			p.stepServerAddAppLink(),
+			p.stepServerConfigureAppLink1(),
+			p.stepServerConfigureAppLink2(),
+
+			p.stepConnect(),
+			p.stepConnected(),
+
+			p.stepWebhook(),
+			p.stepWebhookDone(),
+
+			p.stepCancel(),
+			p.stepDone(),
+		).
+		InitHTTP(p.gorillaRouter)
+}
+
+var cancelButton = flow.Button{
+	Name:    "Cancel",
+	Color:   flow.ColorDanger,
+	OnClick: flow.Goto(stepCancel),
+}
+
+func continueButton(next flow.Name) flow.Button {
+	return flow.Button{
+		Name:    "Continue",
+		Color:   flow.ColorPrimary,
+		OnClick: flow.Goto(next),
+	}
+}
+
+func skipButton(next flow.Name) flow.Button {
+	return flow.Button{
+		Name:    "DEBUG Skip",
+		Color:   flow.ColorWarning,
+		OnClick: flow.Goto(next),
+	}
+}
+
+func (p *Plugin) stepWelcome() flow.Step {
+	return flow.NewStep(stepSetupWelcome).
+		WithPretext("##### :wave: Welcome to Jira integration! [Learn more](https://github.com/mattermost/mattermost-plugin-jira#readme)").
+		WithTitle("Configure the integration.").
+		WithText("Just a few steps to go!\n" +
+			"- **Step 1:** <>/<> TODO: describe the steps....\n").
+		WithButton(continueButton(stepDelegate)).
+		WithButton(cancelButton)
+}
+
+func (p *Plugin) stepDelegate() flow.Step {
+	return flow.NewStep(stepDelegate).
+		WithPretext("##### :hand: Are you a Jira administrator?").
+		WithText("Configuring the integration requires administrator access to Jira. If you aren't a Jira admin you can ask another Mattermost user to do it.").
+		WithButton(flow.Button{
+			Name:    "Continue myself",
+			Color:   flow.ColorPrimary,
+			OnClick: flow.Goto(stepChooseEdition),
+		}).
+		WithButton(flow.Button{
+			Name:  "Ask someone else",
+			Color: flow.ColorDefault,
+			Dialog: &model.Dialog{
+				Title:       "Send instructions to",
+				SubmitLabel: "Send",
+				Elements: []model.DialogElement{
+					{
+						DisplayName: "", // TODO: This will still show a *
+						Name:        "aider",
+						Type:        "select",
+						DataSource:  "users",
+						Placeholder: "Search for people",
+					},
+				},
```

Comment:
Please use the wording from https://github.com/mattermost/mattermost-plugin-github/blob/146ec2d8d156076c71934d1263eac31a27759976/server/plugin/flows.go#L189-L199